### ISSUE TITLE
Building Nix Binary from Source | Autoconf Command Failed

### ISSUE THREAD

**Issue by rowhit** [NixOS/nix#4100]
Hi,

I am trying to build nix from source to use a non-default nix store path. However, I get the following when I try to run the bootstrap.sh script inside the nix source dir.

```errror
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force 
autoreconf: configure.ac: tracing
autoreconf: configure.ac: not using Libtool
autoreconf: running: /nix/store/ljslqprzcgm569kg2bvcrc65nl97gy3x-autoconf-2.64/bin/autoconf --force
configure.ac:83: error: possibly undefined macro: AC_DEFINE
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
configure.ac:110: error: possibly undefined macro: AC_CHECK_HEADERS
configure.ac:118: error: possibly undefined macro: AC_MSG_ERROR
configure.ac:187: error: possibly undefined macro: AC_SEARCH_LIBS
autoreconf: /nix/store/ljslqprzcgm569kg2bvcrc65nl97gy3x-autoconf-2.64/bin/autoconf failed with exit status: 1
``` 
Here is the list of packages I have in my environment
```
autoconf-2.64
autoconf-archive-2019.01.06
automake-1.16.2
bash-4.4-p23
bison-3.7.1
brotli-1.0.7
bzip2-1.0.6.0.1
curl-7.72.0
flex-2.6.4
gcc-wrapper-9.3.0
git-minimal-2.28.0
gnum4-1.4.18
gnumake-4.3
libseccomp-2.5.0
libtool-2.4.6
libxml2-2.9.10
libxslt-1.1.34
nix-2.3.6
openssl-1.1.1g
pkg-config-wrapper-0.29.2
sqlite-3.33.0
vim-8.2.1522
xml2-0.5
xz-5.2.5
```

**Comment by rowhit** [comment]
I have been able to build it successfully now. I am not sure what has changed. May be it was an environment issue.

**Comment by lspitzner** [comment]
Just for the future: https://stackoverflow.com/a/37980772 describes how you might get a more specific error message (in my case the error vanished even). I'll ask if this is something to fix over at #3658.

**Comment by krasin** [comment]
I was hit by the same issue; first tried multiple recipes from the Internet, but what worked is installing `autoconf-archive`, as suggested by @lspitzner. Thank you!
